bundles/tampere/content-editor/view/SideContentEditor.js

```diff
@@ -474,48 +477,49 @@ Oskari.clazz.define('Oskari.tampere.bundle.content-editor.view.SideContentEditor
                 requestData.featureFields = featureData;
             }
 
-
             requestData.featureId = (me.operationMode == "edit" && me._getFeatureData().length > 0 ? me._getFeatureData()[0].value : null);
             requestData.layerId = me.selectedLayerId;
             requestData.srsName = this.sandbox.getMap().getSrsName();
             requestData.geometries = {};
             requestData.geometries.data = [];
             if (me.operationMode == "edit" || deleteFeature == true) {
-                me._fillLayerGeometries(requestData.geometries);
+                me._fillLayerGeometries(requestData.geometries, geometries);
             }
-
-            if (geometries != null)
-            {
-                if (geometries.id.indexOf("OpenLayers_Geometry_MultiPoint_") == 0) {
-                    requestData.geometries.type = "multipoint";
-                    for (var i = 0; i < geometries.components.length; i++) {
-                        requestData.geometries.data.push({x: geometries.components[i].x, y: geometries.components[i].y});
-                    }
-                } else if (geometries.id.indexOf("OpenLayers_Geometry_MultiLineString_") == 0) {
-                    requestData.geometries.type = "multilinestring";
-                    for (var i = 0; i < geometries.components.length; i++) {
-                        var tmpLineString = [];
-                        for (var j = 0; j < geometries.components[i].components.length; j++) {
-                            tmpLineString.push({x: geometries.components[i].components[j].x, y: geometries.components[i].components[j].y});
-                        }
-                        requestData.geometries.data.push(tmpLineString);
-                    }
-                } else if (geometries.id.indexOf("OpenLayers_Geometry_MultiPolygon_") == 0) {
-                    requestData.geometries.type = "multipolygon";
-                    for (var i = 0; i < geometries.components.length; i++) {
-                        var tmpPolygon = [];
-                        for (var j = 0; j < geometries.components[i].components.length; j++) {
-                            var tmpLinearString = [];
-                            for (var k = 0; k < geometries.components[i].components[j].components.length; k++) {
-                                tmpLinearString.push({x: geometries.components[i].components[j].components[k].x, y: geometries.components[i].components[j].components[k].y});
+            if (geometries != null) {
```

Comment:
What's going on in here? Please add some comment why this is done. A similar loop is implemented in _fillLayerGeometries() so maybe you could move this in another function like cloneGeometry() or similar that better describes what's being done.